### PR TITLE
add ietf hardware

### PR DESCRIPTION
**Pull request by mattiaswal**
* Fix coding style in statd.
* Reduce code duplication in statd.
* Represent the VPD as a component in ietf-hardware

### CODE REVIEW

File: src/statd/statd.c [hunk 1]
```diff
@@ -566,6 +572,12 @@ int main(int argc, char *argv[])
 		sr_disconnect(sr_conn);
 		return EXIT_FAILURE;
 	}
+	err = sub_to_hardware(&statd);
```
Comment (rical) [inline src/statd/statd.c:575]: I think it's time to make all these subscriptions calls into a function. Something like `int subscribe(&statd)` which does all these `sub_to` calls without the `sr_disconnect()` in each.

**Review by rical — COMMENTED** [review]
Looks good. Only cosmetic comments from me.

File: src/statd/statd.c (hunk 1 above)
Comment (mattiaswal) [reply src/statd/statd.c:575]: Yes, it is a good thing for the future to do that.

File: src/statd/statd.c [hunk 2]
```diff
@@ -129,7 +129,7 @@ static json_t *json_get_ip_link(void)
 }
 
 static int ly_add_yanger_data(const struct ly_ctx *ctx, struct lyd_node **parent,
-			      char *model, const char *arg)
+                              char *model, const char *arg)
```
Comment (rical) [inline src/statd/statd.c]: These still look incorrect to me?

Checkpatch.pl also complains:
```
git show 211921 | ./output/build/linux-6.5.11/scripts/checkpatch.pl

ERROR: code indent should use tabs where possible                                                                                                                                        
#34: FILE: src/statd/statd.c:132:                                                                                                                                                        
+                              char *model, const char *arg)$
```

Comment (mattiaswal) [reply src/statd/statd.c]: i

> These still look incorrect to me?
> 
> Checkpatch.pl also complains:
> 
> ```
> git show 211921 | ./output/build/linux-6.5.11/scripts/checkpatch.pl
> 
> ERROR: code indent should use tabs where possible                                                                                                                                        
> #34: FILE: src/statd/statd.c:132:                                                                                                                                                        
> +                              char *model, const char *arg)$
> ```
That is a result of emacs linux-c-mode i think
Tab-with is set to 8 and there is 6 spaces, 

> These still look incorrect to me?
> 
> Checkpatch.pl also complains:
> 
> ```
> git show 211921 | ./output/build/linux-6.5.11/scripts/checkpatch.pl
> 
> ERROR: code indent should use tabs where possible                                                                                                                                        
> #34: FILE: src/statd/statd.c:132:                                                                                                                                                        
> +                              char *model, const char *arg)$
> ```
I used emacs with linux-c and our clang-formatter on the code... Will check the formatting script in infix.

**Pull request merged**